src/couchapps/ReqMgr/updates/updaterequest.js

```diff
@@ -79,11 +92,14 @@ function(doc, req) {
     	} else {
     		doc[key] = newValues[key];
     	}
-       
+       	
         // If key is RequestStatus, also update the transition
         if (key == "RequestStatus") {
         	updateTransition();
         }
+        
+        updateTaskStepChain("Task", key, newValues[key]);
+        updateTaskStepChain("Step", key, newValues[key]);
```

Comment:
It is a little ugly to iterate over in javascript. :-) I am not sure how much we will save with less function call. But as far as I know, modern javascript is pretty good at handling that so we probably shouldn't concern that much anymore. (Especially if the code is less cleaner)

```
a = ["Task", "Step"]

for (var i in a) {
    var chainType = a[i];
}

```